test/integration/backward_compatible/sql/ResultTest.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const chai = require('chai');
+chai.should();
+const long = require('long');
+
+const RC = require('../../RC');
+const TestUtil = require('../../../TestUtil');
+const { Client } = require('../../../../');
+const { SqlErrorCode } = require('../../../../lib/sql/SqlErrorCode');
+const { SqlRowMetadataImpl } = require('../../../../lib/sql/SqlRowMetadata');
+const { HazelcastSqlException } = require('../../../../lib/core/HazelcastError');
+
+describe('SqlResultTest', function () {
+    let client;
+    let cluster;
+    let someMap;
+    let mapName;
+    let result;
+
+    before(async function () {
+        TestUtil.markClientVersionAtLeast(this, '4.2');
+        cluster = await RC.createCluster(null, null);
+        await RC.startMember(cluster.id);
+        client = await Client.newHazelcastClient({
+            clusterName: cluster.id
+        });
+    });
+
+    beforeEach(async function () {
+        mapName = TestUtil.randomString(10);
+        someMap = await client.getMap(mapName);
+        await someMap.put(0, 1);
+        await someMap.put(1, 2);
+        await someMap.put(2, 3);
+
+        result = client.getSqlService().execute(`SELECT * FROM ${mapName} WHERE this > ?`, [1]);
+    });
+
+    after(async function () {
+        await RC.terminateCluster(cluster.id);
+        await client.shutdown();
+    });
+
+    afterEach(async function () {
+        await someMap.clear();
+    });
+
+    it('should reject iteration after close()', async function () {
+        await result.close();
+
+        try {
```

Comment:
If we change the `getRejection` method above as I suggested, we could use it here as 
```
const error = await TestUtil.getRejectionReasonOrThrow(async () => {
    for await (const row of result) {};
});
```

and I feel this reads better